kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/utils/qaseIO/TestCaseGenerator.java

```diff
@@ -10,52 +12,142 @@
+                log.info("New test case = '" + testCaseTitle + "' created with id " + caseId);
             }
-            casesApi.createCase(getConfig().projectCode(), caseCreate);
-            log.info("New test case = '" + testCaseTitle + "' created");
         }
     }
 
     @SneakyThrows
     public static HashMap<Long, String> getTestCasesTitleAndId() {
         HashMap<Long, String> map = new HashMap<>();
-        TestCaseListResponse response =
-                casesApi.getCases(getConfig().projectCode(), new Filters().status(Filters.SERIALIZED_NAME_STATUS), 100, 0);
-        for (int i = 0; i < Objects.requireNonNull(Objects.requireNonNull(response.getResult()).getEntities()).size(); i++) {
-            map.put(response.getResult().getEntities().get(i).getId(),
-                    response.getResult().getEntities().get(i).getTitle());
+        boolean getCases = true;
+        int offSet = 0;
+        while (getCases) {
+            getCases = false;
+            TestCaseListResponse response =
+                    casesApi.getCases(getConfig().projectCode(), new Filters().status(Filters.SERIALIZED_NAME_STATUS), 100, offSet);
+            if (Objects.requireNonNull(Objects.requireNonNull(response.getResult()).getEntities()).size() > 0) {
+                for (int i = 0; i < Objects.requireNonNull(Objects.requireNonNull(response.getResult()).getEntities()).size(); i++) {
+                    map.put(response.getResult().getEntities().get(i).getId(),
+                            response.getResult().getEntities().get(i).getTitle());
+                }
+                offSet = offSet + 100;
+                getCases = true;
+            }
         }
         return map;
     }
 
-    public static String getClassName(MethodSource testSource) {
+    private static boolean isCaseIdPresentInQaseIo(Method testMethod) {
+        if (testMethod.isAnnotationPresent(CaseId.class)) {
+            long caseId = testMethod.getAnnotation(CaseId.class).value();
+            HashMap<Long, String> cases = getTestCasesTitleAndId();
+            String title;
+            if (!cases.containsKey(caseId)) {
+                FAILED = true;
+                log.error("The method " + testMethod.getName() + " has wrong @CaseId =" + caseId + " that does not exist in Qase.io. " +
+                        "Please put correct @CaseId");
+                return false;
+            } else {
+                for (Map.Entry<Long, String> map : cases.entrySet()) {
+                    if (map.getKey().equals(caseId)) {
+                        title = map.getValue();
+                        if (!title.matches(generateTestCaseTitle(testMethod))) {
+                            log.error("This CaseId =" + caseId + " belong to test with title = " + title);
+                            return false;
+                        }
+                    }
+                }
+            }
+        }
+        return true;
```

Comment:
why `true` if there's no annotation at all?

Reply:
added verification if annotation is not present